### PR TITLE
PROD-403 Handle priority fee request failure

### PR DESCRIPTION
**Pull request by neel-ds**
### Description
It seems the reveal is occasionally failing due to the `getPriorityFeeEstimate` RPC method call failure, and this issue has regressed despite being resolved with the latest updates.

https://github.com/gator-labs/chomp/blob/5df2c285a899e339314163133b721bd042c395d6/app/queries/getPriorityFeeEstimate.ts#L14-L25

This PR includes handling of RPC call failures by retrying the method twice. If the maximum attempts fail, it will throw an error. Additionally, it logs console errors to help identify the cause of this intermittent behavior.
### What are the steps to test that this code is working?
Better to monitor it and take necessary steps later
### Screen shots or recordings for UI changes
NA

### CODE REVIEW

File: app/utils/solana.ts [hunk 1]
```diff
@@ -40,8 +40,21 @@ export const genBonkBurnTx = async (
...
+  // Verify the estimateFee is not null due to RPC request failure in some cases
+  if (estimateFee === null) {
+    for (let i = 0; i < 2; i++) {
+      estimateFee = await getRecentPrioritizationFees(tx);
+      if (estimateFee !== null) break;
+    }
+    
+    // Throw an error if the calls fail after maximum attempts
+    if (estimateFee === null) {
+      throw new Error("Failed to estimate priority fee after maximum attempts");
```
Comment (yash431garg) [inline app/utils/solana.ts:54]: Instead of throwing an error we can add a fix priority fees incase estimate fee return null.

Comment (neel-ds) [reply app/utils/solana.ts:54]: We can think of a reasonable number.

**Review by marvinmarnold — DISMISSED** [review]

File: app/utils/solana.ts [hunk 2]
```diff
@@ -40,8 +40,21 @@ export const genBonkBurnTx = async (
...
 
-  const estimateFee = await getRecentPrioritizationFees(tx);
+  let estimateFee = await getRecentPrioritizationFees(tx);
 
+  // Verify the estimateFee is not null due to RPC request failure in some cases
+  if (estimateFee === null) {
+    for (let i = 0; i < 2; i++) {
+      estimateFee = await getRecentPrioritizationFees(tx);
+      if (estimateFee !== null) break;
+    }
```
Comment (marvinmarnold) [inline app/utils/solana.ts:50]: Ideally, should sleep here for 2s

**Comment by neel-ds** [comment]
Added median priority fee if maximum attempts reach and delay of 2 secs between re-trials.

cc: @marvinmarnold @yash431garg

**Comment by yash431garg** [comment]
> Added median priority fee if maximum attempts reach and delay of 2 secs between re-trials.
> 
> cc: @marvinmarnold @yash431garg

Looks fine to me how much time does it take incase when we use 5000 as estimatefee

**Comment by neel-ds** [comment]
> > Added median priority fee if maximum attempts reach and delay of 2 secs between re-trials.
> > cc: @marvinmarnold @yash431garg
> 
> Looks fine to me how much time does it take incase when we use 5000 as estimatefee?

6-7 secs

**Review by yash431garg — DISMISSED** [review]

**Review by marvinmarnold — APPROVED** [review]

**Review by yash431garg — APPROVED** [review]

**Pull request merged**